### PR TITLE
Increase code covergae for OverFlowMenuV2

### LINKED ISSUE #17507: Increase `OverflowMenuV2` unit test coverage

### PR DESCRIPTION
**Pull request by soumyaraju**
Closes #17507

Increase code coverage for overflowmenuv2

### CODE REVIEW

**Review by tay1orjones — CHANGES_REQUESTED** [review]
Hey thanks for taking a look at this! Rather than mocking the various components, these tests would be more valuable if they ensured the real code works as expected. Meaning if you `render()` something like what's shown/documented in storybook:

```
<OverflowMenuV2>
  <CustomMenuItem />
  <MenuItem label="Stop app"  />
  <MenuItemDivider />
  <MenuItem label="Delete app" kind="danger" />
</OverflowMenuV2>
```
We'd want to assert that
1. `console.warn` is called as expected with the deprecation warning
2. `useFeatureFlag('enable-v12-overflowmenu')` returns `true` from within `CustomMenuItem`
3. Everything is passed through and spread onto the new implementation of OverflowMenu, meaning new OverflowMenu props such as `autoAlign` are able to be set and have the proper result (auto align class is present) and the children MenuItem are rendered okay like `expect(getByText('Stop app')).toBeInTheDocument();`

This way the only thing you'll need to mock is the implementation of console.warn.

**Comment by soumyaraju** [comment]
Hi @tay1orjones  apologies for the delayed response on this PR. Could you kindly verify if my tests align with your suggestions? Thanks in advance!

File: packages/react/src/components/OverflowMenuV2/OverFlowMenuV2-test.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          <MenuItem label="Stop app" />
+          <MenuItem label="Delete app" kind="danger" />
+        </OverflowMenuV2>
+      </FeatureFlags>
+    );
+
+    const button = screen.getByRole('button', { name: /options/i });
+
+    await act(async () => {
+      fireEvent.click(button);
```
Comment (2nikhiltom) [inline packages/react/src/components/OverflowMenuV2/OverFlowMenuV2-test.js]: fireEvent can be replaced with userEvent. 
`await userEvent.click(button)`

File: packages/react/src/components/OverflowMenuV2/OverFlowMenuV2-test.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright IBM Corp. 2016, 2024
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { act, fireEvent, render, screen } from '@testing-library/react';
```
Comment (2nikhiltom) [inline packages/react/src/components/OverflowMenuV2/OverFlowMenuV2-test.js]: fireEvent can be replaced by userEvent
import userEvent from '@testing-library/user-event';

File: packages/react/src/components/OverflowMenuV2/OverFlowMenuV2-test.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+import { FeatureFlags } from '../FeatureFlags';
+import { MenuItem } from '../Menu';
+import { OverflowMenuV2 } from './';
+import React from 'react';
+import { action } from '@storybook/addon-actions';
+
+// Mocking console.warn to track deprecation warning
+jest.spyOn(console, 'warn').mockImplementation(() => {});
```
Comment (2nikhiltom) [inline packages/react/src/components/OverflowMenuV2/OverFlowMenuV2-test.js]: I think we can remove this along with the commented line on top

File: packages/react/src/components/OverflowMenuV2/OverFlowMenuV2-test.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+    // Check if the deprecation warning is logged
+    expect(consoleWarnSpy).toHaveBeenCalledWith(
+      expect.stringContaining(
+        '`<OverflowMenuV2>` is deprecated and will be removed in the next major version.'
+      )
+    );
+  });
+
+  it('renders with FeatureFlags and passes the correct flag', async () => {
```
Comment (2nikhiltom) [inline packages/react/src/components/OverflowMenuV2/OverFlowMenuV2-test.js]: Could we change test names for consistency? and try something like below
"should render correctly with feature flag enabled"

File: packages/react/src/components/OverflowMenuV2/OverFlowMenuV2-test.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    await act(async () => {
+      const button = getByRole('button', { name: /options/i });
+      button.click();
+    });
+
+    expect(await findByText('Stop app')).toBeInTheDocument();
+    expect(await findByText('Delete app')).toBeInTheDocument();
+  });
+
+  it('passes new props to OverflowMenu and applies autoAlign', () => {
```
Comment (2nikhiltom) [inline packages/react/src/components/OverflowMenuV2/OverFlowMenuV2-test.js]: "should respect align prop "

**Comment by soumyaraju** [comment]
Hey @2nikhiltom, thank you for reviewing it, could you let me know what changes require?

File: packages/react/src/components/OverflowMenuV2/OverFlowMenuV2-test.js (hunk 5 above)
Comment (soumyaraju) [reply packages/react/src/components/OverflowMenuV2/OverFlowMenuV2-test.js]: Done

File: packages/react/src/components/OverflowMenuV2/OverFlowMenuV2-test.js (hunk 4 above)
Comment (soumyaraju) [reply packages/react/src/components/OverflowMenuV2/OverFlowMenuV2-test.js]: Updated test names accordingly

File: packages/react/src/components/OverflowMenuV2/OverFlowMenuV2-test.js (hunk 3 above)
Comment (soumyaraju) [reply packages/react/src/components/OverflowMenuV2/OverFlowMenuV2-test.js]: Thank you, removed it

File: packages/react/src/components/OverflowMenuV2/OverFlowMenuV2-test.js (hunk 2 above)
Comment (soumyaraju) [reply packages/react/src/components/OverflowMenuV2/OverFlowMenuV2-test.js]: Done

**Comment by soumyaraju** [comment]
Thanks @2nikhiltom for the quick review, i have made the changes, please verify it.

**Review by 2nikhiltom — APPROVED** [review]

**Review by ariellalgilmore — APPROVED** [review]

**Review by guidari — APPROVED** [review]

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**